pkg/controllers/user/systemimage/systemimage.go

```diff
@@ -0,0 +1,187 @@
+package systemimage
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/rancher/norman/types/convert"
+	"github.com/rancher/rancher/pkg/image"
+	"github.com/rancher/rancher/pkg/settings"
+	"github.com/rancher/types/apis/apps/v1beta2"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+const (
+	systemImageVersionAnn = "field.cattle.io/systemImageVersion"
+	systemProjectAnn      = "management.cattle.io/system-namespace"
+)
+
+var systemProjectLabels = labels.Set(map[string]string{"authz.management.cattle.io/system-project": "true"})
+
+type Syncer struct {
+	clusterName      string
+	daemonsets       v1beta2.DaemonSetInterface
+	daemonsetLister  v1beta2.DaemonSetLister
+	deployments      v1beta2.DeploymentInterface
+	deploymentLister v1beta2.DeploymentLister
+	projectLister    v3.ProjectLister
+	projects         v3.ProjectInterface
+}
+
+func (s *Syncer) Sync(key string, obj *v3.Project) error {
+	projects, err := s.projectLister.List(s.clusterName, systemProjectLabels.AsSelector())
+	if err != nil {
+		return fmt.Errorf("get projects failed, %v", err)
+	}
+
+	var systemProject *v3.Project
+	for _, v := range projects {
+		if v.Spec.DisplayName == "System" {
+			systemProject = v.DeepCopy()
+		}
+	}
+
+	if systemProject == nil {
+		return nil
+	}
+
+	latestSysImageVersion, err := GetSystemImageVersion()
```

Comment:
I feel like this check can be done in the very beginning. This way we can save on some calls if get system image version fails